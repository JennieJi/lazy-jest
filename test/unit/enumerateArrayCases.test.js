const enumerateArrayCases = require('../../src/caseGenerator/enumerateArrayCases');

describe('enumerateArrayCases', () => {
  test('single arg, valid cases only', () => {
    expect(enumerateArrayCases([
      {
        name: 'a',
        validCases: [1, 2]
      }
    ])).toMatchSnapshot();
  });

  test('single arg, has invalid cases', () => {
    expect(enumerateArrayCases([
      {
        name: 'a',
        validCases: [1, 2],
        invalidCases: [null]
      }
    ])).toMatchSnapshot();
    expect(enumerateArrayCases([
      {
        name: 'a',
        validCases: [1, 2],
        invalidCases: [null, NaN]
      }
    ], 0)).toMatchSnapshot();
  });

  test('multiple arga, valid cases only', () => {
    expect(enumerateArrayCases([
      {
        name: 'a',
        validCases: [1, 2],
      }, {
        name: 'b',
        validCases: ['haha']
      }
    ])).toMatchSnapshot();
  });

  test('multiple arga, has invalid case', () => {
    expect(enumerateArrayCases([
      {
        name: 'a',
        validCases: [1, 2],
        invalidCases: [null, NaN]
      }, {
        name: 'b',
        validCases: ['haha', 'test']
      }
    ], 0)).toMatchSnapshot();
  });
});