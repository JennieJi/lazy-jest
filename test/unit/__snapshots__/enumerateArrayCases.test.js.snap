// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`enumerateArrayCases multiple arga, has invalid case 1`] = `
Array [
  Array [
    null,
    "haha",
  ],
  Array [
    NaN,
    "haha",
  ],
]
`;

exports[`enumerateArrayCases multiple arga, valid cases only 1`] = `
Array [
  Array [
    1,
    "haha",
  ],
  Array [
    2,
    "haha",
  ],
]
`;

exports[`enumerateArrayCases single arg, has invalid cases 1`] = `
Array [
  Array [
    1,
  ],
  Array [
    2,
  ],
]
`;

exports[`enumerateArrayCases single arg, has invalid cases 2`] = `
Array [
  Array [
    null,
  ],
  Array [
    NaN,
  ],
]
`;

exports[`enumerateArrayCases single arg, valid cases only 1`] = `
Array [
  Array [
    1,
  ],
  Array [
    2,
  ],
]
`;
